platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultConfigurationCache.kt

```diff
@@ -268,76 +271,88 @@ class DefaultConfigurationCache internal constructor(
             )
-            ConfigurationCacheAction.STORE
+            logBootstrapSummary(description)
+            ConfigurationCacheAction.STORE to description
         }
 
         else -> {
             when (val checkedFingerprint = checkFingerprint()) {
                 is CheckedFingerprint.NotFound -> {
-                    logBootstrapSummary(
-                        "{} as no cached configuration is available for {}",
+                    val description = formatBootstrapSummary(
+                        "%s as no cached configuration is available for %s",
                         buildActionModelRequirements.actionDisplayName.capitalizedDisplayName,
                         buildActionModelRequirements.configurationCacheKeyDisplayName.displayName
                     )
-                    ConfigurationCacheAction.STORE
+                    logBootstrapSummary(description)
+                    ConfigurationCacheAction.STORE to description
                 }
 
                 is CheckedFingerprint.EntryInvalid -> {
-                    logBootstrapSummary(
-                        "{} as configuration cache cannot be reused because {}.",
+                    val description = formatBootstrapSummary(
+                        "%s as configuration cache cannot be reused because %s.",
                         buildActionModelRequirements.actionDisplayName.capitalizedDisplayName,
                         checkedFingerprint.reason
                     )
-                    ConfigurationCacheAction.STORE
+                    logBootstrapSummary(description)
+                    ConfigurationCacheAction.STORE to description
                 }
 
                 is CheckedFingerprint.ProjectsInvalid -> {
-                    logBootstrapSummary(
-                        "{} as configuration cache cannot be reused because {}.",
+                    val description = formatBootstrapSummary(
+                        "%s as configuration cache cannot be reused because %s.",
                         buildActionModelRequirements.actionDisplayName.capitalizedDisplayName,
                         checkedFingerprint.reason
                     )
-                    ConfigurationCacheAction.UPDATE
+                    logBootstrapSummary(description)
+                    ConfigurationCacheAction.UPDATE to description
                 }
 
                 is CheckedFingerprint.Valid -> {
-                    logBootstrapSummary("Reusing configuration cache.")
-                    ConfigurationCacheAction.LOAD
+                    val description = StructuredMessage.forText("Reusing configuration cache.")
+                    logBootstrapSummary(description)
+                    ConfigurationCacheAction.LOAD to description
                 }
             }
         }
     }
 
+    private
+    fun formatBootstrapSummary(message: String, vararg args: Any?) =
+        StructuredMessage.forText(String.format(Locale.US, message, *args))
```

Comment:
🤔 This could even move to `StructuredMessage.format(message: String, vararg args: Any?)` for discoverability

Reply:
I generally prefer not to expose the `String.format` contract on strings in APIs, even if they are internal. It's mostly okay here, because the utility is local.